crates/bevy_physics/src/d3/intersectors.rs

```diff
@@ -0,0 +1,94 @@
+use super::{geometry::*, ray::*};
+use glam::Vec3;
+
+pub struct RayHit {
+    t: f32,
+    point: Vec3,
+}
+
+impl RayHit {
+    pub fn new(t: f32, point: Vec3) -> Self {
+        Self { t, point }
+    }
+
+    pub fn t(&self) -> &f32 {
+        &self.t
+    }
+
+    pub fn point(&self) -> &Vec3 {
+        &self.point
+    }
+}
+
+pub trait RayIntersector {
+    fn intersect_ray(&self, ray: &Ray) -> Option<RayHit>;
+}
+
+impl RayIntersector for Plane {
+    fn intersect_ray(&self, ray: &Ray) -> Option<RayHit> {
+        let d = self.normal().dot(*ray.direction());
+        if d.abs() > f32::EPSILON {
+            let t = (*self.center() - *ray.origin()).dot(*self.normal()) / d;
+            if t > 0.0 {
```

Comment:
I am guessing the t and d is taken from some formula. We can give it a more verbose name in our code.

Reply:
I would agree and have changed the names.